clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1276,7 +1276,12 @@ void CodeGenModule::EmitCtorList(CtorList &Fns, const char *GlobalName) {
     ctor.addInt(Int32Ty, I.Priority);
     ctor.add(llvm::ConstantExpr::getBitCast(I.Initializer, CtorPFTy));
     if (I.AssociatedData)
-      ctor.add(llvm::ConstantExpr::getBitCast(I.AssociatedData, VoidPtrTy));
+      if (I.AssociatedData->getType()->getPointerAddressSpace() !=
+          VoidPtrTy->getAddressSpace())
+        ctor.add(
+            llvm::ConstantExpr::getAddrSpaceCast(I.AssociatedData, VoidPtrTy));
```

Comment:
You are adding a new address space cast, but I don't see any address space casts in the test.

Reply:
Thanks, I will modify the test to show this.